### DISCUSSION TITLE
metric-history pagination demo

### DISCUSSION THREAD

**Discussion by harupy** [mlflow/mlflow#7491] [General]
Code: https://github.com/mlflow/mlflow/pull/7495

https://user-images.githubusercontent.com/17039389/206600716-69c9cd87-6d01-4a13-90ef-e54d2f4587b2.mov

**Comment by harupy** [comment]
Change the plot style while fetch metrics:


https://user-images.githubusercontent.com/17039389/206678879-a0ae2c9c-742c-424c-94b0-98bf480ac428.mov

**Reply by BenWilson2** [reply]
Awesome! Nothing to worry about with this then :)

**Comment by harupy** [comment]
Fetch a new metric:

https://user-images.githubusercontent.com/17039389/206679375-186884b0-70ea-4848-9f6d-d7f6ed0fd6fd.mov

**Comment by harupy** [comment]
Fetch multiple metrics:

https://user-images.githubusercontent.com/17039389/206679589-37f6f688-be3a-4315-bcac-11fda306f009.mov

**Comment by harupy** [comment]
Fetch another metric when other metric is still being fetched:

https://user-images.githubusercontent.com/17039389/206680373-94cdce7d-34e1-449d-b679-3db2d461acce.mov

**Reply by harupy** [reply]
The spinner disappears when `x` is exhausted, but it should not because `y` is NOT exhausted yet.

**Reply by BenWilson2** [reply]
So long as the binding for displaying the spinner applies to all elements associated with the plot this should be a fairly straight forward fix, right?

**Comment by harupy** [comment]
@BenWilson2 @dbczumar Added a few more demos.

**Comment by harupy** [comment]
Toggle a metric:

https://user-images.githubusercontent.com/17039389/206682147-5c1e3e27-dd42-4b0f-8ead-9e32be1331c6.mov

**Reply by BenWilson2** [reply]
What happens when a metric is toggled on, then off, then on again while pagination queries are in progress?
